### DISCUSSION TITLE
Issues when using the beta (native) Android client

### DISCUSSION THREAD

**Discussion by dfunkt** [dani-garcia/vaultwarden#4798] [Q&A]
Sometimes when I log in the beta app these errors pop up in the logs:
```
vaultwarden  | [2024-07-29 03:05:46.340][vaultwarden::auth][ERROR] Token has expired
vaultwarden  | [2024-07-29 03:05:46.624][auth][ERROR] Unauthorized Error: Invalid claim
vaultwarden  | [2024-07-29 03:05:46.642][vaultwarden::api::core::accounts::_][WARN] Request guard `Headers` failed: "Invalid claim".
```
No problems with the regular app, web vault or browser extension.
Posting this here to see if this happens to anyone else (or maybe it's something specific to my setup).

**Comment by dfunkt** [comment]
It almost feels like, for some reason, the access_token isn't being refreshed so it ends up expiring. But I don't see anything wrong in the Bitwaren dev APK logs:
```
09-01 14:40:38.567 13475 25685 D BitwardenNetworkClient: --> POST https://xx/identity/connect/token
09-01 14:40:38.567 13475 25685 D BitwardenNetworkClient: Content-Type: application/x-www-form-urlencoded
09-01 14:40:38.567 13475 25685 D BitwardenNetworkClient: Content-Length: 148
09-01 14:40:38.567 13475 25685 D BitwardenNetworkClient: User-Agent: Bitwarden_Mobile/2024.8.1 (debug/standard) (Android 13; SDK 33; Model M2012K11AG)
09-01 14:40:38.567 13475 25685 D BitwardenNetworkClient: Bitwarden-Client-Name: mobile
09-01 14:40:38.567 13475 25685 D BitwardenNetworkClient: Bitwarden-Client-Version: 2024.8.1
09-01 14:40:38.567 13475 25685 D BitwardenNetworkClient: Device-Type: 0
09-01 14:40:38.568 13475 25685 D BitwardenNetworkClient: client_id=mobile&refresh_token=5q7I..%3D%3D&grant_type=refresh_token
09-01 14:40:38.568 13475 25685 D BitwardenNetworkClient: --> END POST (148-byte body)
09-01 14:40:38.776 13475 25685 D BitwardenNetworkClient: <-- 200 https://xx/identity/connect/token (207ms)
```

**Comment by krasen1234** [comment]
Having the same issue, it started after the new UI and crap with the latest version of bitwarden for Android. Only happens on Android not the iPhone app or browser apps.

**Comment by krasen1234** [comment]
It results in I have to login each 1-2 days.

**Reply by BlackDex** [reply]
If a claim is invalid then something either with date/time is not correct on either the client or server.
Or something in between is messing with the data like reverse proxies, or a WAF like CloudFlare for example.

Check the `/admin/diagnostics` to see if there are any date/time issues.
Also check https://time.is via your mobile device to check and see if you are not out-off-sync with date/time.

**Reply by krasen1234** [reply]
Hello,
Been using the same setup for years now and just started to have issues from the last Android version. Are you using it?

**Reply by BlackDex** [reply]
Yes, I'm using it and have no issues.
Still claims have nothing to do with new clients, those are tokens which are time-sensitive, and will fail if the date/time is not correct. Stating it is working fine for years doesn't mean your host, reverse proxy, WAF etc... might have something changed too, or has a wrong date/time for example for some reason.

There is always a chance of something happening with anything in between which can cause issues.
Only blaming the end-product is not always right.

To debug stuff you need to work either backwards or forwards checking each step in-between the client and server and see where items are maybe modified or filtered.
